letsencrypt-compatibility-test/letsencrypt_compatibility_test/interfaces.py

```diff
@@ -0,0 +1,57 @@
+"""Let's Encrypt compatibility test interfaces"""
+import zope.interface
+
+import letsencrypt.interfaces
+
+# pylint: disable=no-self-argument,no-method-argument
+
+
+class IPluginProxy(zope.interface.Interface):
+    """Wraps a Let's Encrypt plugin"""
+    def add_parser_arguments(cls, parser):
+        """Adds command line arguments needed by the parser"""
+
+    def __init__(args):
+        """Initializes the plugin with the given command line args"""
+
+    def cleanup_from_tests():
+        """Performs any necessary cleanup from running plugin tests.
+
+        This is guarenteed to be called before the program exits.
+
+        """
+
+    def has_more_configs():
+        """Returns True if there are more configs to test"""
+
+    def load_config():
+        """Loads the next config and returns its name"""
+
+
+class IConfiguratorBaseProxy(IPluginProxy):
+    """Common functionality for authenticator/installer tests"""
+    http_port = zope.interface.Attribute(
+        "The port to connect to on localhost for HTTP traffic")
+
+    https_port = zope.interface.Attribute(
+        "The port to connect to on localhost for HTTPS traffic")
+
+    def get_testable_domain_names():
+        """Returns the domain names that can be used in testing"""
+
+
+class IAuthenticatorProxy(
```

Comment:
nit: The IConfiguratorBaseProxy is a bit confusing. (Throughout letsencrypt configurator implies authenticator and installer).  Can the IPluginProxy and IConfiguratorBaseProxy be combined?  (Is there ever a IPluginProxy that is not a IConfiguratorBaseProxy?)